mypy/types.py

```diff
@@ -1778,6 +1778,7 @@ class CallableType(FunctionLike):
         # (this is used for error messages)
         "imprecise_arg_kinds",
         "unpack_kwargs",  # Was an Unpack[...] with **kwargs used to define this callable?
+        "erased",  # Is this callable created as an erased form of a more precise type?
```

Comment:
Yeah. This looks quite ad-hoc. The more  think about it the more I think we should do this unconditionally. This will make the whole thing much simpler.